### ISSUE TITLE
Is there any way to control loss proportions among multiple tasks?

### ISSUE THREAD

**Issue by huhk-sysu** [UKPLab/sentence-transformers#927]
Hi.

Say I'm training models in a multi-task learning way, and I pass 2 training objectives to `tentenceTransformer.fit`. How can I control the loss proportions like below?

```
loss = 0.8 * loss_1 + 0.2 * loss_2
```

**Comment by nreimers** [comment]
Currently not. The two datasets are used alternatively, i.e., the system first does a step along loss1, then a step along loss2.

Datasets with different sizes is no issue. The dataloader starts from start if one dataset runs out of examples.

**Comment by alejandrojcastaneira** [comment]
I noticed that in multi-task training the loaded per epoch its limited data to the size of the smaller dataset, so some samples of the other datasets possibly will not be shown to the models during training, its there a way to control this behaviour, eg. once one dataloader run of out samples during an epoch to continue loading the samples from the others?

**Comment by nreimers** [comment]
Hi @alejandrojcastaneira 
That is correct.

What you can do is to increase the number of epochs (or control it via the number of steps parameter).

E.g. 
dataset1: 5k examples
dataset2: 10k examples

Train for 1 epoch => 5k examples from dataset1 and 5k examples from dataset2 are used
Train for 2 epoch => two times 5k examples from dataset1 and all 10k examples from dataset2 are used.

**Comment by alejandrojcastaneira** [comment]
If my understanding it's correct the torch dataloader should continue on the remaining samples from previous iterations?

if there its a big scale difference between the dataset sizes, could the model overfit on the smaller dataset?

**Comment by nreimers** [comment]
Yes and yes
